skills_ml/algorithms/embedding/base.py

```diff
@@ -17,115 +18,114 @@ class Word2VecModel(object):
-            s3_conn (:obj: `boto.s3.connection.S3Connection`): the boto object to connect to S3.
-            files (:obj: `list` of (str)): model files need to be downloaded/loaded.
+            storage (:obj: `skills_ml.Store`): skills_ml Store object
             model (:obj: `gensim.models.doc2vec.Doc2Vec`): gensim doc2vec model.
-            lookup (dict): lookup table for mapping each jobposting index to soc code.
         """
+        self.storage = storage
+        self._model = model
+        self.model_name = None
+
+    @property
+    def storage(self):
+        return self._storage
+
+    @storage.setter
+    def storage(self, value):
+        if value.__class__.__name__ in [c.__name__ for c in Store.__subclasses__()]:
+            self._storage = value
+        else:
+            raise Exception(f"{value} is not Store Object")
+
+    def load_model(self, model_name):
+        """The method to load the model from where Storage object specified
+
+        model_name (str): name of the model to be used.
+        """
+        self._model = self.storage.load(model_name)
         self.model_name = model_name
-        self.s3_path = s3_path + self.model_name
-        self.s3_conn = s3_conn
-        self.model = self._load_model() if model == None else model
 
-    def _load_model(self):
-        """The method to download the model from S3 and load to the memory.
+    def write_model(self, model_name):
+        """The method to write the model to where the Storage object specified
 
-        Returns:
-            gensim.models.doc2vec.Doc2Vec: The word-embedding model object.
+        model_name (str): name of the model to be used.
         """
-        files  = list_files(self.s3_conn, self.s3_path)
-        with tempfile.TemporaryDirectory() as td:
-            for f in files:
-                filepath = os.path.join(td, f)
-                if not os.path.exists(filepath):
-                    logging.info('calling download from %s to %s', self.s3_path + f, filepath)
-                    download(self.s3_conn, filepath, os.path.join(self.s3_path, f))
-            model = Word2Vec.load(os.path.join(td, self.model_name+".model"))
+        self.storage.write(self._model, model_name)
 
-            return model
+    def inference(self):
+        raise NotImplementedError
 
 
 class Doc2VecModel(object):
     """The Doc2VecModel Object is a base object which specifies which word-embeding model.
 
     Example:
     ```
-    from airflow.hooks import S3Hook
     from skills_ml.algorithms.embedding.base import Doc2VecModel
 
-    s3_conn = S3Hook().get_conn()
-    doc2vec_model = Doc2VecModel(s3_conn=s3_conn)
+    doc2vec_model = Doc2VecModel()
+    doc2vec_model.load_model(path)
     ```
     """
-    def __init__(self, model_name='doc2vec_2017-07-14T11:32:00.997426',
-        lookup=None, model=None, s3_conn=None, s3_path=S3_PATH_EMBEDDING_MODEL):
+    def __init__(self, storage=FSStore(), model=None, lookup=None):
```

Comment:
I think we don't want to actually instantiate an FSStore here as a default argument for the same reason we don't want to use lists as default arguments. We should leave this as None and just instantiate the FSStore inside the body of __init__ if it is None.